models/dodo/ethereum/dodo_ethereum_schema.yml

```diff
@@ -0,0 +1,100 @@
+version: 2
+
+models:
+  - name: dodo_ethereum_trades
+    meta:
+      blockchain: ethereum
+      sector: dex
+      project: dodo
+      contributors: scoffie
+    config:
+      tags: ['ethereum','dodo','trades', 'dex']
+    description: >
+        Dodo trades on Ethereum.
+    tests:
+      - dbt_utils.unique_combination_of_columns:
+          combination_of_columns:
+            - block_date
+            - blockchain
+            - project
+            - version
+            - category
+            - tx_hash
+            - evt_index
+            - trade_id
+    columns:
+      - &blockchain
+        name: blockchain
+        description: "Blockchain which the DEX is deployed"
+      - &project
+        name: project
+        description: "Project name of the DEX"
+      - &version
+        name: version
+        description: "Version of the contract built and deployed by the DEX project"
+      - &category
+        name: category
+        description: "Type of route through which a trade happened. Through DEX contract or and Aggregator contract"
+      - &block_date
+        name: block_date
+        description: "Truncated to day UTC event block date of each DEX trade"
+      - &block_time
+        name: block_time
+        description: "UTC event block time of each DEX trade"
+      - &token_bought_symbol
+        name: token_bought_symbol
+        description: "Token symbol for token bought in the trade"
+      - &token_sold_symbol
+        name: token_sold_symbol
+        description: "Token symbol for token sold in the trade"
+      - &token_pair
+        name: token_pair
+        description: "Token symbol pair for each token involved in the trade"
+      - &token_bought_amount
+        name: token_bought_amount
+        description: "Value of the token bought at time of execution with its decimals factored in"
+      - &token_sold_amount
+        name: token_sold_amount
+        description: "Value of the token sold at time of execution with its decimals factored in"
+      - &token_bought_amount_raw
+        name: token_bought_amount_raw
+        description: "Raw value of the token bought at time of execution without division by its decimals"
+      - &token_sold_amount_raw
+        name: token_sold_amount_raw
+        description: "Raw value of the token sold at time of execution without division by its decimals"
+      - &amount_usd
+        name: amount_usd
+        description: "USD value of the trade at time of execution"
+      - &token_bought_address
```

Comment:
please apply the generic test for token_bought on this field, example:
https://github.com/duneanalytics/spellbook/blob/main/models/uniswap/ethereum/uniswap_ethereum_schema.yml#L64

in order for it to run, values need manually added to the seed file:
https://github.com/duneanalytics/spellbook/blob/main/seeds/dex/trades/dex_trades_seed.csv

most dexes have analytics sites that link tx's from the dex to etherscan, where you can grab the necessary values to populate the seed. you can also query the source decoded table for that tx_hash to find the values to put in place.

the goal is to ensure we aren't flipping token bought/sold on the fields in the model

Reply:
Gotcha